python/cudf/cudf/tests/test_concat.py

```diff
@@ -587,3 +587,80 @@ def test_concat_dataframe_with_multiIndex(df1, df2):
     actual = pd.concat([pdf1, pdf2], axis=1)
 
     assert_eq(expected, actual)
+
+
+@pytest.mark.parametrize("ignore_index", [True, False])
+@pytest.mark.parametrize("sort", [True, False])
+def test_concat_inner_join(ignore_index, sort):
+    pdf1 = pd.DataFrame(
+        {
+            "x": range(10),
+            "y": list(map(float, range(10))),
+            "z": list("abcde") * 2,
+        }
+    )
+    pdf2 = pd.DataFrame(
+        {"x": range(10, 20), "y": list(map(float, range(10, 20)))}
+    )
+    pdf3 = pd.DataFrame({"v": [1, 2], "x": [1, 2], "y": [1, 2], "z": [1, 2]})
+
+    gdf1 = gd.from_pandas(pdf1)
+    gdf2 = gd.from_pandas(pdf2)
+    gdf3 = gd.from_pandas(pdf3)
+    # Make empty frame
+    gdf_empty1 = gdf2[:0]
+    assert len(gdf_empty1) == 0
+    pdf_empty1 = gdf_empty1.to_pandas()
+
+    assert_eq(
+        pd.concat(
+            [pdf1, pdf2], sort=sort, join="inner", ignore_index=ignore_index
+        ),
+        gd.concat(
+            [gdf1, gdf2], sort=sort, join="inner", ignore_index=ignore_index
+        ),
+    )
+
+    assert_eq(
+        pd.concat(
+            [pdf1, pdf2, pdf3, pdf_empty1],
+            sort=sort,
+            join="inner",
+            ignore_index=ignore_index,
+        ),
+        gd.concat(
+            [gdf1, gdf2, gdf3, gdf_empty1],
+            sort=sort,
+            join="inner",
+            ignore_index=ignore_index,
+        ),
+    )
+
+    assert_eq(
+        pd.concat([pdf1], sort=sort, join="inner", ignore_index=ignore_index),
+        gd.concat([gdf1], sort=sort, join="inner", ignore_index=ignore_index),
+    )
+
+    s1 = gd.Series(["a", "b", "c"])
+    s2 = gd.Series(["a", "b"])
+    s3 = gd.Series(["a", "b", "c", "d"])
+    s4 = gd.Series()
+    ps1 = s1.to_pandas()
+    ps2 = s2.to_pandas()
+    ps3 = s3.to_pandas()
+    ps4 = s4.to_pandas()
+
+    assert_eq(
+        gd.concat(
+            [s1, s2, s3, s4],
+            sort=sort,
+            join="inner",
+            ignore_index=ignore_index,
+        ),
+        pd.concat(
+            [ps1, ps2, ps3, ps4],
+            sort=sort,
+            join="inner",
+            ignore_index=ignore_index,
+        ),
+    )
```

Comment:
```suggestion
@pytest.mark.parametrize("ignore_index", [True, False])
@pytest.mark.parametrize("sort", [True, False])
@pytest.mark.parametrize("join", ["inner", "outer"])
def test_concat_join(ignore_index, sort, join):
    pdf1 = pd.DataFrame(
        {
            "x": range(10),
            "y": list(map(float, range(10))),
            "z": list("abcde") * 2,
        }
    )
    pdf2 = pd.DataFrame(
        {"x": range(10, 20), "y": list(map(float, range(10, 20)))}
    )
    pdf3 = pd.DataFrame({"v": [1, 2], "x": [1, 2], "y": [1, 2], "z": [1, 2]})

    gdf1 = gd.from_pandas(pdf1)
    gdf2 = gd.from_pandas(pdf2)
    gdf3 = gd.from_pandas(pdf3)
    # Make empty frame
    gdf_empty1 = gdf2[:0]
    assert len(gdf_empty1) == 0
    pdf_empty1 = gdf_empty1.to_pandas()

    assert_eq(
        pd.concat(
            [pdf1, pdf2], sort=sort, join=join, ignore_index=ignore_index
        ),
        gd.concat(
            [gdf1, gdf2], sort=sort, join=join, ignore_index=ignore_index
        ),
    )

    assert_eq(
        pd.concat(
            [pdf1, pdf2, pdf3, pdf_empty1],
            sort=sort,
            join=join,
            ignore_index=ignore_index,
        ),
        gd.concat(
            [gdf1, gdf2, gdf3, gdf_empty1],
            sort=sort,
            join=join,
            ignore_index=ignore_index,
        ),
    )

    assert_eq(
        pd.concat([pdf1], sort=sort, join=join, ignore_index=ignore_index),
        gd.concat([gdf1], sort=sort, join=join, ignore_index=ignore_index),
    )

    s1 = gd.Series(["a", "b", "c"])
    s2 = gd.Series(["a", "b"])
    s3 = gd.Series(["a", "b", "c", "d"])
    s4 = gd.Series()
    ps1 = s1.to_pandas()
    ps2 = s2.to_pandas()
    ps3 = s3.to_pandas()
    ps4 = s4.to_pandas()

    assert_eq(
        gd.concat(
            [s1, s2, s3, s4],
            sort=sort,
            join=join,
            ignore_index=ignore_index,
        ),
        pd.concat(
            [ps1, ps2, ps3, ps4],
            sort=sort,
            join=join,
            ignore_index=ignore_index,
        ),
    )
```

Can we parametrize `join`? Since this is a new parameter that we are adding support to, we should be testing it both ways.

Reply:
This sounds good to me Prem! I'll commit these changes

Reply:
Just noticed that there was a build error. I hope it's not anything with the tests. I'll check on it again later today.

Reply:
It looks like a style issue but the tests are passing on my end :/ I'll look through this